addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,899 @@
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return commandList;
+    }
+
+    private String[] getRemoteList(int index) {
+
+        String getRemotes = "Agetremotes " + index + (char) 13;
+        ByteBuffer response = sendCommand(getRemotes);
+        String[] remoteList = new String[0];
+
+        if (response != null) {
+            String message = stripByteCount(response).split("\0")[0];
+            logger.trace("remotes returned {}", message);
+            if (message != null) {
+                if (message.contains("REMOTELIST")) {
+                    remoteList = message.split(",");
+                } else {
+                    logger.warn("Received some non-compliant command ({})", message);
+                    disconnect(socketChannel);
+                }
+            }
+        } else {
+            logger.warn("Did not receive an answer from the IRtrans transceiver for '{}' - Parsing is skipped",
+                    getThing().getUID());
+            disconnect(socketChannel);
+        }
+
+        return remoteList;
+    }
+
+    private ByteBuffer sendCommand(String command) {
+
+        if (command != null) {
+            ByteBuffer byteBuffer = ByteBuffer.allocate(command.getBytes().length);
+            try {
+                byteBuffer.put(command.getBytes("ASCII"));
+                onWritable(byteBuffer, socketChannel);
+                Thread.sleep(((BigDecimal) getConfig().get(RESPONSE_TIME_OUT)).intValue());
+                return onReadable(socketChannel, ((BigDecimal) getConfig().get(BUFFER_SIZE)).intValue());
+            } catch (UnsupportedEncodingException | NumberFormatException | InterruptedException e) {
+                Thread.currentThread().interrupt();
+                logger.error("An exception occurred while configurting the IRtrans transceiver for '{}': {}",
+                        getThing().getUID(), e.getMessage());
+            }
+        }
+
+        return null;
+    }
+
+    @SuppressWarnings("null")
+    private SocketChannel connect(String ipAddress, int portNumber) {
+        SocketChannel socketChannel = null;
+        try {
+            socketChannel = SocketChannel.open();
+            socketChannel.socket().setKeepAlive(true);
+            socketChannel.configureBlocking(false);
+
+            synchronized (selector) {
+                selector.wakeup();
+                int interestSet = SelectionKey.OP_READ | SelectionKey.OP_WRITE | SelectionKey.OP_CONNECT;
+                socketChannelKey = socketChannel.register(selector, interestSet);
+            }
+
+            InetSocketAddress remoteAddress = new InetSocketAddress(ipAddress, portNumber);
+            logger.trace("Connecting the channel for {} ", remoteAddress);
+            socketChannel.connect(remoteAddress);
+        } catch (IOException e) {
+            logger.error("An exception occurred while connecting connecting to '{}:{}' : {}", ipAddress, portNumber,
```

Comment:
double "connecting".
So what? Is this a bug in the code or shouldn't you rather set the Thing status?